lua/cfc_weapon_lockouts/server/sv_locker.lua

```diff
@@ -0,0 +1,224 @@
+        CFCWeaponLockouts._lockWarns[ply] = {}
+    end )
+end )
+
+hook.Add( "PlayerDroppedWeapon", "CFC_WeaponLockouts_LockWeapon", function( ply, wep )
+    CFCWeaponLockouts.lockWeapon( ply, wep, true )
+end )
+
+hook.Add( "EntityRemoved", "CFC_WeaponLockouts_LockWeapon", function( ent )
+    if not IsValid( ent ) or not ent:IsWeapon() then return end
+
+    CFCWeaponLockouts.lockWeapon( ent )
+end )
+
+hook.Add( "PlayerSwitchWeapon", "CFC_WeaponLockouts_TrackWeaponOwner", function( ply, old, new )
+    if not IsValid( new ) then return end
+
+    new.weaponLockoutOwner = ply
+end )
+
+hook.Add( "WeaponEquip", "CFC_WeaponLockouts_CanPickup", function( wep, ply )
+    if not IsValid( wep ) or not IsValid( ply ) then return end
+
+    local weaponClass = wep:GetClass()
+    ply.weaponLockoutWeapons = ply.weaponLockoutWeapons or {}
+    wep.weaponLockoutOwner = ply
+
+    local plyWeapons = ply.weaponLockoutWeapons
+    local isLocked = CFCWeaponLockouts.weaponIsLocked( ply, weaponClass )
+
+    -- In certain cases, EntityRemoved gets called after WeaponEquip, causing odd behavior where the weapon is unlocked, meant to be locked, and not held by the player.
+    -- Manually keeping track of the weapon classes held by a player allows us to catch that error.
+    if not isLocked and plyWeapons[weaponClass] and not CFCWeaponLockouts.NOT_LOCKABLE[wep:GetClass()] then
+        isLocked = true
+        CFCWeaponLockouts.lockWeapon( ply, wep, false )
+    end
+
+    plyWeapons[weaponClass] = true
+
+    if not isLocked then
+        CFCWeaponLockouts._lockWarns[ply][weaponClass] = "unlocked"
+        CFCWeaponLockouts._lockWarns[ply].unlockedCount = ( CFCWeaponLockouts._lockWarns[ply].unlockedCount or 0 ) + 1
+
+        return
+    end
+
+    local identifier = "CFC_WeaponLockouts_LockWarn_" .. ply:SteamID()
+    local warns = CFCWeaponLockouts._lockWarns[ply] or {}
+    warns[weaponClass] = "locked"
+    warns.lockedCount = ( warns.lockedCount or 0 ) + 1
+
+    timer.Create( identifier, CFCWeaponLockouts.WARN_BUILDUP:GetFloat(), 1, function()
+        warnPlayer( identifier, ply, warns )
+    end )
+
+    local primaryAmmoType = wep:GetPrimaryAmmoType()
+    local secondaryAmmoType = wep:GetSecondaryAmmoType()
+    local primaryAmmo = ply:GetAmmoCount( primaryAmmoType ) or 0
+    local secondaryAmmo = ply:GetAmmoCount( secondaryAmmoType ) or 0
+    local clip1 = math.min( wep:Clip1() or 0, wep:GetMaxClip1() or 0 )
+    local clip2 = math.min( wep:Clip2() or 0, wep:GetMaxClip2() or 0 )
+
+    -- Empties ammo and clips from the locked weapon
+    timer.Simple( 0, function()
+        if not IsValid( wep ) then return end
+        if primaryAmmo > 0 then ply:SetAmmo( 0, primaryAmmoType ) end
+        if secondaryAmmo > 0 then ply:SetAmmo( 0, secondaryAmmoType ) end
+        if clip1 > 0 then wep:SetClip1( 0 ) end
+        if clip2 > 0 then wep:SetClip2( 0 ) end
+    end )
```

Comment:
If a `0s` timer is necessary could you add a comment explaining why, so that we're not afraid to remove it later if we can fix it?